test/functional/test_framework/p2p.py

```diff
@@ -600,6 +600,19 @@ def test_function():
 
     # Message receiving helper methods
 
+    def _check_last_getdata(self, hash_list, datatype=None):
+        last_data = self.last_message.pop("getdata", None)
```

Comment:
Note: This will pop from last message via `check_last_getdata()` call as well, which is different from previous commits. That's why I accepted `last_data` as an argument in my suggestion.
Is popping in `check_last_getdata` also fine?

Reply:
Yeah, I noticed that on your suggestion, but I don't think we do reuse the value in any test (same as with `wait_for_*`)